syntax = "proto3";

package brokerservice;

service BrokerService{

 rpc SendTransaction(Transaction) returns (Response) {}
 rpc GetUpdates(Request) returns (Queries) {}

}

message Response{
 bytes data = 1;  // dict response
}

message Transaction{
 bytes data = 1;  // dict transaction: 
                  // {'req': 'Enqueue', 'pid': 12, 'topic': 'foo', 'partition': 'B', 'message': 'test'}
}

message Request{}

message Queries{
 repeated string queries = 1;
}
